crates/subspace-service/src/dsn_archiver.rs

```diff
@@ -0,0 +1,55 @@
+use futures::StreamExt;
+use sc_consensus_subspace::{ArchivedSegmentNotification, SubspaceLink};
+use sp_core::traits::SpawnEssentialNamed;
+use sp_core::Encode;
+use sp_runtime::traits::Block as BlockT;
+use subspace_networking::PUB_SUB_ARCHIVING_TOPIC;
+use tracing::{error, info, trace};
+
+/// Start an archiver that will listen for archived segments and send it to DSN network using
+/// pub-sub protocol.
+pub fn start_subspace_dsn_archiver<Block>(
+    subspace_link: &SubspaceLink<Block>,
+    node_config: subspace_networking::Config,
+    spawner: &impl SpawnEssentialNamed,
+) where
+    Block: BlockT,
+{
+    spawner.spawn_essential_blocking(
+        "subspace-archiver-DSN",
+        None,
+        Box::pin({
+            trace!("DSN archiver started.");
+            let mut archived_segment_notification_stream = subspace_link
+                .archived_segment_notification_stream()
+                .subscribe();
+
+            async move {
+                let (node, node_runner) = subspace_networking::create(node_config).await.unwrap();
```

Comment:
`unwrap()` is an artifact from the early prototype - I changed it to `match` with error logging.

Reply:
I would not separate those tasks because it would require introducing some async data channel between them which is IMHO overkill for this method.

Reply:
I don't see why synchronization is needed between them. Runner and archiver are already running in two separate background tasks, the only difference is that if network creation fails, it will fail in background tasks and crash an app later instead of not letting an app start fully in the first place.